File: src/SignalR/common/SignalR.Common/src/Protocol/PingMessage.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace Microsoft.AspNetCore.SignalR.Protocol;

/// <summary>
/// A keep-alive message to let the other side of the connection know that the connection is still alive.
/// </summary>
public class PingMessage : HubMessage
{
    /// <summary>
    /// A static instance of the PingMessage to remove unneeded allocations.
    /// </summary>
    public static readonly PingMessage Instance = new PingMessage();

    private PingMessage()
    {
    }
}

File: src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/PingMessage.java
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

package com.microsoft.signalr;

public class PingMessage extends HubMessage
{
    private final int type = HubMessageType.PING.value;

    private static PingMessage instance = new PingMessage();

    private PingMessage()
    {
    }

    public static PingMessage getInstance() {return instance;}

    @Override
    public HubMessageType getMessageType() {
        return HubMessageType.PING;
    }
}